modules/vstudio/vs2010_vcxproj.lua

```diff
@@ -3214,3 +3402,189 @@
+			else
+				m.element("DebugInformationFormat", nil, "Minimal")
+			end
+
+		end
+	end
+
+	function m.linuxExceptionHandling(cfg, condition)
+		if cfg.exceptionhandling then
+			m.element("ExceptionHandling", condition,  iif(cfg.exceptionhandling == p.OFF, "Disabled", "Enabled"))
+		end
+	end
+
+	function m.linuxFloatingPointModel(cfg)
+
+		if cfg.floatingpoint then
+			m.element("RelaxIEEE", nil, iif(cfg.floatingpoint == p.OFF, "false", "true"))
+		end
+
+	end
+
+	function m.linuxLanguageStandardCpp(cfg)
+		local cpp_langmap = {
+			["C++98"]   = "c++98",
+			["C++03"]   = "c++98",
+			["C++11"]   = "c++11",
+			["C++14"]   = "c++14",
+			["C++17"]   = "c++17",
+			["C++2a"]   = "c++2a",
+			["C++20"]   = "c++20",
+			["C++latest"] = "c++2a",
+			["gnu++98"] = "gnu++98",
+			["gnu++03"] = "gnu++03",
+			["gnu++11"] = "gnu++11",
+			["gnu++14"] = "gnu++14",
+			["gnu++17"] = "gnu++17",
+			["gnu++17"] = "gnu++20",
+		}
+		
+		if cpp_langmap[cfg.cppdialect] ~= nil then
+			m.element("CppLanguageStandard", nil, cpp_langmap[cfg.cppdialect])
+		end
+	end
+
+	function m.linuxLanguageStandardC(cfg)
+		local c_langmap = {
+			["C89"]   = "c89",
+			["C99"]   = "c99",
+			["C11"]   = "c11",
+			["gnu99"] = "gnu99",
+			["gnu11"] = "gnu11",
+		}
+
+		if c_langmap[cfg.cdialect] ~= nil then
+			m.element("CLanguageStandard", nil, c_langmap[cfg.cdialect])
+		end
+	end
+
+	function m.remoteDeployDir(cfg)
+		if cfg.remotedeploydir ~= nil then
+			m.element("RemoteDeployDir", nil, cfg.remotedeploydir)
+		end
+	end
+
+	function m.remoteProjectRelDir(cfg)
+		if cfg.remoteprojectrelativedir ~= nil then
+			m.element("RemoteProjectRelDir", nil, cfg.remoteprojectrelativedir)
+		end
+	end
+
+	function m.remoteProjectDir(cfg)
+		if cfg.remoteprojectdir ~= nil then
+			m.element("RemoteProjectDir", nil, cfg.remoteprojectdir)
+		end
+	end
+
+	function m.remoteRootDir(cfg)
+		if cfg.remoterootdir ~= nil and cfg.remoterootdir ~= "" then
+			m.element("RemoteRootDir", nil, cfg.remoterootdir)
+		end
+	end
+
+	function m.linuxPIC(cfg, condition)
+		if cfg.pic ~= nil then
+			m.element("PositionIndependentCode", condition, iif(cfg.pic == "On", "true", "false"))
+		end
+	end
+
+	function m.linuxPlatformToolset(cfg)
+		local tool, version = p.config.toolset(cfg)
+
+		if not version then
+			local value = p.action.current().toolset
+			tool, version = p.tools.canonical(value)
+		end
+
+		local gcc_map = {
+			["remote"] = "Remote_GCC_1_0",
+			["wsl"] = "WSL_1_0",
+			["wsl2"] = "WSL2_1_0",
+		}
+		
+		local clang_map = {
+			["remote"] = "Remote_Clang_1_0",
+			["wsl"] = "WSL_Clang_1_0",
+			["wsl2"] = "WSL2_Clang_1_0",
+		}
+
+		if cfg.toolchainversion then
+
+			local map = iif(cfg.toolset == "gcc", gcc_map, clang_map)
+			version  = map[cfg.toolchainversion]
+
+		end
+
+		if version then
+			m.element("PlatformToolset", nil, version)
+		else
+			p.error('Invalid toolchainversion for the selected toolset (%s).', cfg.toolset)
```

Comment:
```suggestion
			p.error('Invalid toolchainversion (%s) for the selected toolset (%s).', cfg.toolchainversion, cfg.toolset)
```
?

Reply:
@nickclark2016 This I am still a bit unsure of. Do you have any particular opinion on it? What is the difference between toolsversion and toolchainversion?